### ISSUE TITLE
Non-deterministic results from 'decode_msg'

### ISSUE THREAD

**Issue by bwallinspire** [M0r13n/pyais#40]
I am seeing some non-determinism in `decode_msg` output, e.g.:

```
>>> raw = b'!AIVDM,1,1,,A,D02=ag0flffp,0*11'
>>> pyais.decode_msg(raw)['offset2']
2
>>> pyais.decode_msg(raw)['offset2']
0
```

Memory overflow? This AIS message itself seems short but I would expect an error if it was unexpected length?

python: 3.9
pyais: 1.6.2
bitarray: 2.3.4

**Comment by M0r13n** [comment]
@bwallinspire Thank you for bringing this up. That’s a weird issue. I will take a look at it as soon as possible.

**Comment by M0r13n** [comment]
@bwallinspire 

I could reproduce your error. It reason is, that I missed quite a few length checks. Therefore, the lib was reading data out of bounds. I am not quite sure, why the `bitarray` module allowed access to data outside the array, but I added length checks.

Now, the lib returns `None` for fields that are outside the message. All unittests succeed, but I need to please mypy. This will be done tomorrow. :-) See PR #41 

Again, thanks for bringing this up!

**Comment by M0r13n** [comment]
@bwallinspire 

You can now install the fixed version `1.6.3`:

`pip install pyais==1.6.3`
